added a getting started guide

masukomi opened the pull request:
* replaces getting started content in README.md with link to new
  getting_started.md file
* adds new Getting Started entry to top level sidebar.
* adds a bunch of screenshots to help newbs.

-- 

I'm open to suggestions for rewording or expanding any of this.

ahdinosaur commented on README.md:
```diff
@@ -19,35 +19,10 @@
 
-## How To Get Started
+## Ready to get started? 
+Check out our [Getting Started Guide](getting_started.md)
```
given that the "Getting Started Guide" is now much more involved than the previous getting started guide, would it make sense to keep both, one on the front page as a _quick start_, then link to the "Scuttlebutt newbie guide" or something for more detail? although i can see us wanting to share more context before they leave the website, i'm concerned that people will see the "Guide" and gfto.

masukomi replied:
yes, and no. I think there's value in a "Quick Start" guide, but i think the existing one is a bit _too_ quick. 

I'll take a stab at expanding the current one. Maybe have this section be 

```
## Ready to get started?
 
Check out the [Getting Started Guide](getting_started.md), 
or if you're in a rush you can use the [Quick Start Guide](quick_start.md). 
```

Also have the Getting started guide link to the quick start guide at the top for people who need it fast.

ahdinosaur replied:
i think the quick start guide should be on the front page, it's so quick you don't need to click a link and wait for another page to load.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+
+As for the social network, there are many clients, just like there are many Twitter clients. It doesn't really matter which one you use. They're all talking on the same network. 
+
+Patchwork is the most polished client and it doesn't require you to know any geeky stuff. The only caveat is that it just wants to run on one computer, and your identity is tied to the files on the computer you install it on. There are ways around this but it's beyond the scope of this.
```
"The only caveat is that it just wants to run on one computer, and your identity is tied to the files on the computer you install it on. There are ways around this but it's beyond the scope of this." this is an opinion baked into the Scuttlebutt protocol, not Patchwork. as mentioned before it's not by accident. %7QPjGHfSvJMogZWDb1Mffbs7K9MwOt+INC8Z7A6V7Fc=.sha256

masukomi replied:
so, would you suggest saying "The only caveat is scuttlebutt wants to only run on one computer..." ?  I think it's important to call out the only run it on one computer thing because that is _not_ normal and most people would expect to be able to run it on their work and home computers... and when we get phones they'll expect to be able to use the same identity there too (maybe with `sameAs` but still...).

ahdinosaur replied:
Scuttlebutt the protocol expects one key per uxer per device, so replication can be simple. if we don't already have a faq section on this, we should, since it's probably the most commonly asked question. however it doesn't mean you can't use Scuttlebutt apps on all your devices, `sameAs` will give us simple replication _and_ multi-device feeds.

masukomi replied:
for now though, i can't find any documentation on how to use sameAs or what the real consequences of using it are, so i'm not sure what to put here. :(

Also if it's cli only i'm not sure it should be mentioned here.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+
+
+## Step 0
+If you haven't done so already go to [the Scuttlebutt Home Page](README.md) and watch the video. Then come back. 
```
serious ask for a 10 minute video intro! or maybe totally reasonable 😄

masukomi replied:
not sure how to interpret that statement.

ahdinosaur replied:
sorry. i think expecting the reader to watch a 10 minute video is another indicator that this is not a quick start guide. but i think it makes sense as a detailed getting started guide!

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+If you haven't done so already go to [the Scuttlebutt Home Page](README.md) and watch the video. Then come back. 
+
+## Step 1
+[Download Patchwork](http://dinosaur.is/patchwork-downloader/), and install it like any other app. 
```
"install it like any other app" might sound really demoralizing when you download a ".AppImage" file and are really confused what to do with it, or any other operating system and get confused at what to do with what you download.

masukomi replied:
I'm not sure what it's like on windows or linux, but on the mac it's literally just double click the thing you downloaded and drag the Patchwork icon into the Applications folder just like the icons show you.  It's _very_ simple and is just like any other app (that isn't from the app store). 

Is it just my experience talking that makes me think that's trivial? 
I can expand on it (double-click the .dmg, etc...) but i would need someone else to provide instructions for windows and linux installs.

ahdinosaur replied:
i understand it's trivial, i just mean the language used might be harmful _in the rare case where someone is confused now to install the app_. i think Windows is an .exe, which is about as common as it comes, so maybe the only confusing case is the new shiny AppImages for Linux. for macOS, do you _run_ the installer? might be worth saying, run the downloaded installer, or something. or yeah, we could give a short phrase on how to install for each os.

masukomi replied:
> for macOS, do you run the installer? 

no. for macOS there is no installer. there's a .dmg file (disk image) which you double click and it mounts it and opens up a window that looks like this: 

<img width="540" alt="patchwork_3_3_0" src="https://user-images.githubusercontent.com/2829/34399339-88dbe668-eb54-11e7-9eb7-d0befd6687f4.png">

That's the actual application on the left and a symlink to your Applications folder on the right. You drag the app into the folder and you're done.  This is also how mac apps have been installed for probably over a decade now (not counting getting them from the app store).

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+
+It doesn't matter what name you put. It can be your real name. It can be a nickname. Like the real world it's entirely possible for multiple people to have the same name. Your friends can give you nicknames too. 
+
+You don't need to upload an image, but I recommend that you do, because it's much easier to figure out who is talking when there's a visual, and the defaults are really generic. Some people use a photo of themselves. Some people use a random avatar they like, just like Twitter.
```
nice

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+
+You don't need to upload an image, but I recommend that you do, because it's much easier to figure out who is talking when there's a visual, and the defaults are really generic. Some people use a photo of themselves. Some people use a random avatar they like, just like Twitter.
+
+The text area, like every text area in Patchwork uses [markdown](http://commonmark.org/help/) formatting.
```
might be good to list some special [Scuttlebutt markdown (`ssb-markdown`)](https://github.com/ssbc/ssb-markdown) things?

masukomi replied:
yeah, i _want_ to do that, and i intend to change it eventually, but so far the only docs I've found are _terrible_ looking and absolutely not something you want to show a new user unfamiliar with markdown.  Maybe there's a pretty version with visualized formatting. For example: actually showing a link not just saying "`[link text](link_url)` makes a link". If so i haven't found it.  I figured do this first, then tackle those docs, then update this link.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+## Step 3 
+
+Get connected.  
+Because Scuttlebutt is a distributed network you need to find some people to follow. This is partially because it's really _really_ boring if you're not listening to anyone and no-one is listening to you. It's also because you get "gossip" from the people you follow. "Gossip" is more than just info about people you're interested in. It is also the name of [a protocol](https://en.wikipedia.org/wik ...
```
maybe link to "gossip" topic on the handbook

masukomi replied:
ahh yes, better.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+
+## Getting all the Gossip
+
+As soon as you connect to a pub or start following a person Scuttlebutt will start syncing down all the "gossip" they have for you, and pass on any "gossip" you have for them. *The first time you connect to a pub server there will be a lot to download. * So, give it some time.
```
'*' formatting problem? or at least it confused the GitHub markdown renderer.

masukomi replied:
weird. i guess it's the lack of space before the first one? dunno. should be valid. Will correct.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+
+![browse all button](assets/patchwork-browse-all.png)
+
+Find a topic that looks interesting and start reading. When you're ready to "Publish" remember that it uses Markdown formatting. 
```
i wonder if it's worth mentioning that channels won't increase your replication scope, as in you won't access any new messages by subscribing to channels, just one's you already have will show up in the interface.

masukomi replied:
I feel like replication scope is a topic that _definitely_ needs a guide for new users, and should be linked to from here but is probably too much to be included here. I'm still getting my head around what exactly the limits are so I'm not currently qualified to write that.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+Profiles are a little different. A profile is a feed for a single person. So, in that case you want to use "Copy Link Ref" For example, if I wanted to link to André's profile I'd say `[here's André's profile](@QlCTpvY7p9ty2yOFrv1WU1AE88aoQc4Y7wYal7PFc+w=.ed25519)`  
+
+### Sharing Scuttlebutt things with other people
+I wanted to share a really interesting conversation with a friend of mine who wasn't on Scuttlebutt yet, so I right clicked on the first message in the thread, chose "Copy External Link" and then messaged them that link. You could Tweet it or post it on Facebook, or whatever.  The link goes to a public site where you can see, but not interact with, posts on Scuttlebutt.
```
maybe mention this is all possible by the good grace of @clehner hosting a central server. eventually hope that more pubs will have external portals by default.

masukomi replied:
dunno, feels a little off topic. I've got no trouble giving someone props for something like that but it seems a little odd to put here (to me). You folks own the repo. You get to make the call.

side note: i looked into setting one up but saw no docs on it. I saw a commit message that looked like that ability had been added to the pub server, but wasn't sure if that was correct and if so how to turn it on, or if it was just automatically on by default if you set up a pub server or....  People don't create docs for the tools they write and then wonder why others don't start using them... :/

ahdinosaur replied:
sorry, i meant less of giving props, more of setting expectations for why the external portal running on a single vps might be down.

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+### Sharing Scuttlebutt things with other people
+I wanted to share a really interesting conversation with a friend of mine who wasn't on Scuttlebutt yet, so I right clicked on the first message in the thread, chose "Copy External Link" and then messaged them that link. You could Tweet it or post it on Facebook, or whatever.  The link goes to a public site where you can see, but not interact with, posts on Scuttlebutt.
+
+It should be noted that people can indicate that they don't want their stuff shown on public sites or in search results on sites like Google. 
```
link to this? otherwise feel like i'm missing something.

masukomi replied:
i read that this was a feature that this had been recently added by @arj03 ( and saw multiple references to this functionality on ssb ) but i have found no documentation on it. I'd love to add it in here, because _i_ haven't figured out how to do it.

arj03 replied:
http://between-two-worlds.dk:8807/%25MvSSfCGqCmWjaod%2Fm08b22poYMi8bWOVfS4r%2BlLok%2Bc%3D.sha256

ahdinosaur replied:
@masukomi yeah sorry, i have seen this, i meant it would be good to link to it (thanks @arj03!), because a reader might be frustrated by not being given a link for more context.

masukomi commented on getting_started.md:
```diff
@@ -0,0 +1,104 @@
+Because Scuttlebutt is a distributed network you need to find some people to follow. This is partially because it's really _really_ boring if you're not listening to anyone and no-one is listening to you. It's also because you get "gossip" from the people you follow. "Gossip" is more than just info about people you're interested in. It is also the name of [a protocol](https://en.wikipedia.org/wik ...
+
+### Connecting Locally
+Most of us don't happen to be sitting next to someone else on the network. If you _are_ you'll see them listed under "local" on the left. Click their profile to see more about them, and click "Follow" if you feel like following what that person has to say (just like Twitter).  You'll start downloading information from them that they've posted to the network _and_ information their friends have po ...
```
@glowkeeper pointed out that he didn't see the "Local" which means this isn't clear enough about it only being visible IF someone else is running ssb on your local network.

masukomi commented:
@ahdinosaur 

I think I've addressed all the comments except the ones about the ssb markdown (as noted) and the one about distance of what you can see (as noted). 

There's a quickstart in the README.md now (Please comment)

I've updated the FAQ to now have a privacy page with info about the `--publicWebHosting` / `--no-publicWebHosting` options. I also took the bit about choosing whatever name you wanted out of the tor.md and linked to tor.md from within privacy.md

NOTE: on my computer gitbook is refusing to render privacy.md, but it was also refusing to render tor.md before i started poking those files too so... who knows. Gitbook isn't the most ... bug free codebase in my experience.

arj03 commented:
Good splitting up tor and privacy, that is an excellent idea. Thanks!

ahdinosaur commented on README.md:
```diff
@@ -19,35 +19,28 @@
+4. Click "+ Join Pub" in the top left corner of Patchwork and paste in the
+   invite code.  It's best to have multiple pub servers to extend your
+   visibility into the network. 
+5. Start browsing public posts and channels. Note that there's a "More Channels"
```
do we want to say anything about following new people? i noticed that some newbies have been subscribing to many channels, then being confused why they are mostly empty, since they are still only following a single pub.

i guess i mean to say: i'd personally rather we talk about following people than subscribing to channels.

masukomi replied:
adding a line between 5 and 6 about following people

ahdinosaur commented on getting_started.md:
```diff
@@ -0,0 +1,166 @@
+
+
+## Step 0 If you haven't done so already go to [the Scuttlebutt Home
+Page](README.md) and watch the video. Then come back. 
```
formatting?

ahdinosaur replied:
could be totally fine, confuses the GitHub renderer, but have no idea how it affects GitBook. i see that you're wrapping lines at a max character length, which makes sense. the same markdown confusion happens again a couple times later below.

masukomi replied:
nope. i reformatted the page before saving and didn't realize it'd do that to the headings with no line break between them and the 1st line. it'd screw up any md renderer and make it all a heading.

ahdinosaur commented:
> NOTE: on my computer gitbook is refusing to render privacy.md, but it was also refusing to render tor.md before i started poking those files too so... who knows. Gitbook isn't the most ... bug free codebase in my experience.

is everything linked up in `SUMMARY.md`? that's a sure way to lose a page.

masukomi commented:
> is everything linked up in SUMMARY.md? that's a sure way to lose a page.

no. none of the sub-pages of FAQ are, but they don't need to be, and i think with the way the sidebar looks right now that's probably appropriate. In my experience Gitbook will (well... should) render all the .md pages it finds even if they aren't linked, but this one definitely is. It's weird.

masukomi commented:
updated
* corrected formatting, 
* added line about following people in quickstart
* added 2 paragraphs about following people in getting started

forgot to mention that, in the last update, i added a note about viewer.scuttlebutt.io being hosted by a community member (to set expectations). Please check the wording on that. --cc @ahdinosaur

masukomi commented:
@moid mentioned 

> One thing I didn't see mentioned that might be helpful is about linking. In Patchwork when you type @.. it offers suggestions of who you would like to reference. Ditto for channels with #..

Let me add a note about that before you merge it.

masukomi commented:
done.

ahdinosaur commented:
> no. none of the sub-pages of FAQ are, but they don't need to be, and i think with the way the sidebar looks right now that's probably appropriate. In my experience Gitbook will (well... should) render all the .md pages it finds even if they aren't linked, but this one definitely is. It's weird.

um, all the FAQ sub-pages are linked up. i've had many bad experiences from forgetting to link pages in SUMMARY.md, so just trying to help.
